src/main/java/com/joyent/manta/monitor/Application.java

```diff
@@ -132,32 +136,60 @@ public static void main(final String[] args) throws InterruptedException {
                 Class<MantaOperationsChain> chainClass =
                         (Class<MantaOperationsChain>)Class.forName(runner.getChainClassName());
                 MantaOperationsChain chain = injector.getInstance(chainClass);
+                String metricPostFix = chain.getClass().getSimpleName();
+                String putRequestLatencyMetric = String.format("manta_monitor_%s_put_request_latency_", testType);
+
+                if ("buckets".equals(testType)) {
+                    client.options(client.getContext().getMantaBucketsDirectory());
+                    if ("FileMultipartUploadGetDeleteChain".equals(metricPostFix)) {
+                        String msg = "Multipart upload not supported for buckets yet."
+                                + "Check the CONFIG_FILE env variable and re-run the application";
+                        LOG.error(msg);
+                        System.exit(1);
+                    }
+                }
+
                 Histogram requestPutHistogram = Histogram.build()
-                        .name("manta_monitor_put_request_latency_seconds_"
-                                + chain.getClass().getSimpleName())
+                        .name(putRequestLatencyMetric + metricPostFix)
                         .help("Metric that gives a cumulative observation for "
-                                + "latency, in seconds, in creating a directory and "
-                                + "uploading a file")
+                                + "latency, in seconds, in creating a directory/bucket and "
+                                + "putting an object")
                         .create();
                 // Add the histogram here, to register it later below
                 requestPutHistogramsMap.put(chain.getClass().getSimpleName(),
                         requestPutHistogram);
                 ChainRunner chainRunner = new ChainRunner(chain, runner,
-                        client, UNCAUGHT_EXCEPTION_HANDLER, requestPutHistogramsMap);
+                        client, UNCAUGHT_EXCEPTION_HANDLER, requestPutHistogramsMap, testType);
                 runningChains.add(chainRunner);
             } catch (ClassNotFoundException e) {
                 LOG.error("Unable to load class: {}", runner.getChainClassName());
+            } catch (IOException e) {
```

Comment:
Do you intend to silently ignore any exceptions that do not meet the criteria you are testing for?

Reply:
@1010sachin were you able to address this concern ?

Reply:
the below comment is the same as above. So yes, it has been addressed.